src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -8661,6 +8661,24 @@ static void sofia_handle_sip_i_state(switch_core_session_t *session, int status,
 				}
 			}
 
+			// Ensure final response is sent on the other leg before processing t38 reinvite
+			if (is_t38) {
+				private_object_t *o_tech_pvt;
+				switch_channel_callstate_t o_callstate;
+
+				if (switch_core_session_get_partner(session, &other_session) == SWITCH_STATUS_SUCCESS) {
+					o_tech_pvt = switch_core_session_get_private(other_session);
+					o_callstate = switch_channel_get_callstate(o_tech_pvt->channel);
+
+					if (o_callstate < CCS_ACTIVE) {
```

Comment:
If we look at the callstate enum
```
	CCS_DOWN,
	CCS_DIALING,
	CCS_RINGING,
	CCS_EARLY,
	CCS_ACTIVE,
	CCS_HELD,
	CCS_RING_WAIT,
	CCS_HANGUP,
	CCS_UNHELD
```
Then value such as CCS_HANGUP and CCS_RING_WAIT can sill be process by your code. CCS_HANGUP can be skipped since the session should not be accessible in hangup. But CCS_RING_WAIT seems to be a possible case here what do you think?

Reply:
Yeah for CCS_HANGUP, I suspect the session won't be active then. For the CCS_RING_WAIT, that seems plausible. I could add a condition for that.

Reply:
Added @dev-ryanc